src/compiler/binder.ts

```diff
@@ -3637,6 +3637,10 @@ namespace ts {
         let excludeFlags = TransformFlags.NodeExcludes;
 
         switch (kind) {

```

Comment:
There's a bug with name conflicts in generated variables in the esnext downlevel: https://github.com/joeywatts/TypeScript/pull/18.
If the fix is big, we can open an issue against this repo and address in a follow-on PR.

Reply:
Resolved joeywatts#18 by changing how the names are generated and reserving the names in nested scopes.
https://github.com/bloomberg/TypeScript/blob/648dc15b677147e7be31756a88ea2b94ddb4f9fb/src/compiler/transformers/esnext.ts#L564-L566